dlib/cmake_utils/test_for_libpng/libpng_test.cpp

```diff
@@ -1,54 +0,0 @@
-// Copyright (C) 2019  Davis E. King (davis@dlib.net), Nils Labugt
```

Comment:
I'm not super jazzed about removing these test compiles.  But we can try it out and see what happens.  The issue is there are an astounding number of people who have broken copies of libraries like this on their systems.  So cmake will think there is a libpng (or libjpeg or whatever), because there is.  But it's some weird edited copy that is missing functions.  For instance, on windows, anaconda is a huge offender.

I don't know if this is still a huge problem anymore since adding all this stuff prevented the issue from arising since this detects such brokenness and automatically shunts to the copy in dlib/external and the windows user goes on happily ignorant of it all.

Although for sure, in a perfect world cmake would filter out that kind of broken stuff automatically and not tell you "hey libpng is available".

Reply:
Yeah I can do that. I mean I would be slowly reverting a lot of what I've done. In general let me know how you feel about the PR. We can choose to ignore it. It was an experiment.

Reply:
Well, it is nice and clean.  I definitely didn't like adding all those gnarly details into dlib's cmake files.  The problem is people expect dlib to Just Work no matter what on every platform.  It is very portable, but then they layer on the fact that they broke their platform somehow, or they installed something that is made by someone who wants to disagree with their platform's standards or how other libraries ought to be.  Like there are packages that literally install copies of libpng with edited headers into paths cmake will find (or even the compiler with default options) and those headers will be missing functions, or have their signatures changed.  Same for X11 headers and other libraries.  Hell, there is a steady stream of people posting errors that indicate that the C++ standard library they have is somehow missing stuff. I've no idea how that happens, but it does.  That's just unrecoverable though.

But maybe that stuff isn't needed anymore with the newer version of cmake.  I also tried some things in ubuntu 18.04 and it all worked like a charm.

Reply:
Anyway, we can leave the compiler testing stuff out, make cmake will take care of all that and/or people have less broken stuff now.  Let's run the experiment and see :)

Reply:
I used https://cmake.org/cmake/help/latest/module/CheckCSourceRuns.html for detecting SIMD intrinsics. I much prefer it to try_compile as you can compile and run source code without separate files. It's really nice. I could add a sprinkle of that for checking libpng and X11 stuff.

Reply:
Yeah CheckCSourceRuns is way nicer than the old way it had to be done.  I think leave the libpng and X11 stuff out.  Let's see how broke or not the world is these days.  Plenty easy to add that stuff back in if reports about it start flooding in again.